lib/utils.ts

```diff
@@ -0,0 +1,67 @@
+import * as filedisk from 'file-disk';
+import type {
+	GetPartitionsResult,
+	GPTPartition,
+	MBRPartition,
+} from 'partitioninfo';
+import { getFsLabel, LabelNotFound } from './fsLabel';
+
+/**
+ * Summarizes the partition found by findPartition().
+ */
+export type FindPartitionResult = {
+	index: number;
+	/** partition name for GPT, or filesystem label for MBR */
+	name: string;
+}
+
+/**
+ * @summary Find the partition on the provided disk with a name that matches one
+ * of the provided names.
+ *
+ * Matches on partition name for GPT partitions or on filesystem label for MBR
+ * partitions.
+ *
+ * @returns A FindPartitionResult with the found index and name/label, or index
+ * of -1 if not found.
+ */
+export async function findPartition(
+	fileDisk: filedisk.FileDisk,
+	partitionInfo: GetPartitionsResult,
+	names: string[],
+): Promise<FindPartitionResult> {
+	const { partitions } = partitionInfo;
+	const isGPT = (
+		partsInfo: GetPartitionsResult,
+		_parts: Array<GPTPartition | MBRPartition>,
+	): _parts is GPTPartition[] => partsInfo.type === 'gpt';
+
+	const result = {index: -1, name: ''};
```

Comment:
I think that since this is a new piece of code, we should use a patterns that fits more with the modern JS/TS patterns.
```suggestion
	let result : FindPartitionResult | undefined;
```